server/graphql/v1/mutations/collectives.js

```diff
@@ -461,17 +466,10 @@ export async function approveCollective(remoteUser, CollectiveId) {
     },
   });
 
-  // Approve all events created by this collective under this host
-  models.Collective.findAll({
-    where: {
-      type: types.EVENT,
-      HostCollectiveId: host.id,
-      ParentCollectiveId: collective.id,
-    },
-  }).then(events => {
-    events.map(event => {
-      event.update({ isActive: true, approvedAt: new Date() });
-    });
+  // Approve all events created by this collective
+  const events = await collective.getEvents();
+  events.map(event => {
```

Comment:
I still feel it's better to `await`, `Promise.all` to parallelize?

Reply:
Yeah, I think that's better, and by what I saw in our DB the number of events is actually pretty small.